Riot/Modules/Onboarding/OnboardingCoordinator.swift

```diff
@@ -264,7 +264,15 @@ final class OnboardingCoordinator: NSObject, OnboardingCoordinatorProtocol {
                 self.authenticationCoordinator(coordinator, didLoginWith: session, and: authenticationFlow, using: authenticationType)
             case .didComplete:
                 self.authenticationCoordinatorDidComplete(coordinator)
-            case .didStart, .clearAllData, .cancel:
+            case .clearAllData:
+                self.showClearAllDataConfirmation {
+                    MXLog.debug("[OnboardingCoordinator] showLegacyAuthenticationScreen: Clear all data after soft logout")
+                    self.authenticationService.reset()
+                    self.authenticationFinished = false
+                    self.cancelAuthentication(flow: .login)
+                    AppDelegate.theDelegate().logoutSendingRequestServer(true, completion: nil)
+                }
```

Comment:
This is the same code as used in the callback from the new flow. @ismailgulek, I was wondering if it would make sense to remove the completion closure and place that code directly in the method to avoid duplication. WDYT?

Reply:
Yeah makes sense to me 👍